jwst/background/background_sub.py

```diff
@@ -26,85 +26,171 @@ def background_sub(input_model, bkg_list, sigma, maxiters):
     bkg_list: filename list
         list of background exposure file names
 
+    sigma: float, optional
+        Number of standard deviations to use for both the lower
+        and upper clipping limits.
+
+    maxiters: int or None, optional
+        Maximum number of sigma-clipping iterations to perform
+
+    bkg_model: JWST data model
+        background data model
+
     Returns
     -------
     result: JWST data model
         background-subtracted target data model
 
     """
+    # Subtract the average background from the member
+    log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
 
     # Compute the average of the background images associated with
     # the target exposure
-    bkg_model = average_background(bkg_list, sigma, maxiters)
+    if input_model.meta.model_type == "ImageModel":
+        data_dim = 2  # rate
+    elif input_model.meta.model_type == "CubeModel":
+        data_dim = 3  # rateints
+    else:
+        log.error("Input target exposure must be either a CubeModel or an ImageModel.")
+        raise ValueError("Expected either a CubeModel or an ImageModel.")
 
-    # Subtract the average background from the member
-    log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
-    result = subtract_images.subtract(input_model, bkg_model)
+    bkg_model = average_background(bkg_list, sigma, maxiters, data_dim)
+
+    if data_dim == 2:  # rate
+        result = subtract_images.subtract(input_model, bkg_model)
+    elif data_dim == 3:  # rateints
+        result = subtract_images.subtract_ints(input_model, bkg_model)
 
-    # We're done. Return the result.
+    # We're done. Return the background model and result.
     return bkg_model, result
 
 
-def average_background(bkg_list, sigma, maxiters):
+def average_background(bkg_list, sigma, maxiters, data_dim):
```

Comment:
don't need `data_dim` to be passed here. All you need to care about in this function is the dimensions of the bkg images, once they get loaded.

Reply:
Resolved this.